pkg/unikontainers/utils_test.go

```diff
@@ -0,0 +1,366 @@
+
+	// Check if the PID file exists
+	_, err = os.Stat(pidFilePath)
+	assert.NoError(t, err, "Expected PID file to exist")
+
+	// Check if the content of the PID file is correct
+	content, err := os.ReadFile(pidFilePath)
+	assert.NoError(t, err, "Expected no error in reading PID file")
+	assert.Equal(t, strconv.Itoa(pid), string(content), "Expected PID file content to be %d", pid)
+
+	// Clean up
+	os.Remove(pidFilePath)
+}
+
+func TestGetInitPid(t *testing.T) {
+	t.Run("init PID found", func(t *testing.T) {
+		t.Parallel()
+		// Create a temporary file for testing
+		tmpFile, err := os.CreateTemp("", "test*.json")
+		assert.NoError(t, err)
+		defer os.Remove(tmpFile.Name())
+
+		// Write test data to the file
+		testData := map[string]interface{}{
+			"init_process_pid": 12345.0,
+		}
+		jsonData, err := json.Marshal(testData)
+		assert.NoError(t, err)
+
+		_, err = tmpFile.Write(jsonData)
+		assert.NoError(t, err)
+		tmpFile.Close()
+
+		// Call the function and check the result
+		pid, err := getInitPid(tmpFile.Name())
+		assert.NoError(t, err, "Expected no error in getting init PID")
+		assert.Equal(t, 12345.0, pid, "Expected PID to be 12345")
+	})
+	t.Run("init PID file not found", func(t *testing.T) {
+		t.Parallel()
+		// Call the function with a non-existent file
+		pid, err := getInitPid("nonexistent.json")
+		assert.Equal(t, float64(0), pid, "Expected PID to be 0 for nonexistent file")
+		assert.NoError(t, err, "Expected no error for nonexistent file")
+	})
+
+	t.Run("init PID invalid JSON", func(t *testing.T) {
+		t.Parallel()
+		// Create a temporary file with invalid JSON
+		tmpFile, err := os.CreateTemp("", "test*.json")
+		assert.NoError(t, err)
+		defer os.Remove(tmpFile.Name())
+
+		_, err = tmpFile.WriteString("{invalid json}")
+		assert.NoError(t, err)
+		tmpFile.Close()
+
+		// Call the function and check the result
+		pid, err := getInitPid(tmpFile.Name())
+		assert.Equal(t, float64(0), pid, "Expected PID to be 0 for invalid JSON")
+		assert.NoError(t, err, "Expected no error for invalid JSON")
+	})
+	t.Run("init PID missing key", func(t *testing.T) {
+		t.Parallel()
+		// Create a temporary file without "init_process_pid"
+		tmpFile, err := os.CreateTemp("", "test*.json")
+		assert.NoError(t, err)
+		defer os.Remove(tmpFile.Name())
+
+		testData := map[string]interface{}{
+			"some_other_key": 12345.0,
+		}
+		jsonData, err := json.Marshal(testData)
+		assert.NoError(t, err)
+
+		_, err = tmpFile.Write(jsonData)
+		assert.NoError(t, err)
+		tmpFile.Close()
+
+		// Call the function and check the result
+		pid, err := getInitPid(tmpFile.Name())
+		assert.Equal(t, float64(0), pid, "Expected PID to be 0 for missing key")
+		assert.NoError(t, err, "Expected no error for missing key")
+	})
+}
+
+func TestCopyFile(t *testing.T) {
+	t.Run("copy file success", func(t *testing.T) {
+		t.Parallel()
+		// Create a temporary source file
+		srcFile, err := os.CreateTemp("", "src*.txt")
```

Comment:
Why do not we use tempDir from testing?

Reply:
I refactored the function to use t.TempDir instead.